CONTRIBUTING.md

```diff
@@ -80,6 +82,22 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
 
 You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `– – signoff` to add the `Signed-off-by` line to the end of the commit message.
 
+## Changelog
+
+OpenSearch-Net maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
+
+Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
+
+
+### How to add my changes to [CHANGELOG](CHANGELOG.md)?
```

Comment:
If you want to frame this as a question from the contributors perspective I'd go with:
```suggestion
### How do I add my changes to the [CHANGELOG](CHANGELOG.md)?
```

Reply:
Otherwise if you want it as a statement of us telling the contributor it'd be more like:
```suggestion
### How to add your changes to the [CHANGELOG](CHANGELOG.md)
```